File: media-service/src/main/java/com/example/media_service/service/MediaService.java
package com.example.media_service.service;

import com.example.media_service.dto.MediaRequestDTO;
import com.example.media_service.dto.MediaResponseDTO;

import java.util.List;

public interface MediaService {
    MediaResponseDTO create(MediaRequestDTO dto);
    MediaResponseDTO getById(Long id);
    List<MediaResponseDTO> getAll();
    List<MediaResponseDTO> getByCategory(String category);
    List<MediaResponseDTO> getByGenre(String genre);
    List<MediaResponseDTO> search(String keyword);
    MediaResponseDTO update(Long id, MediaRequestDTO dto);
    void delete(Long id);
}

File: mediahub-frontend/src/app/features/media/services/media.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { Media } from '../models/media.model';

@Injectable({ providedIn: 'root' })
export class MediaService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = '/api/media';

  getAll(): Observable<Media[]> {
    return this.http.get<Media[]>(this.baseUrl);
  }

  getById(id: number): Observable<Media> {
    return this.http.get<Media>(`${this.baseUrl}/${id}`);
  }

  getByCategory(category: string): Observable<Media[]> {
    const params = new HttpParams().set('category', category);
    return this.http.get<Media[]>(`${this.baseUrl}/search/category`, { params });
  }

  getByGenre(genre: string): Observable<Media[]> {
    const params = new HttpParams().set('genre', genre);
    return this.http.get<Media[]>(`${this.baseUrl}/search/genre`, { params });
  }
}
